example/lib/screen/xayn_widgets/page/dots_indicator_page.dart

```diff
@@ -0,0 +1,304 @@
+        onChanged: (value) {
+          setState(() {
+            dotsSize = value!;
+          });
+        },
+      ),
+    );
+  }
+
+  Widget _buildListTileDotsBorderWidth() {
+    return ListTile(
+      title: const Text('Dots border width: '),
+      trailing: DropdownButton<double>(
+        value: dotsBorderWidth,
+        items: _kDotsBorderWidthList.map(_itemBuilderDotsDimen).toList(),
+        onChanged: (value) {
+          setState(() {
+            dotsBorderWidth = value!;
+          });
+        },
+      ),
+    );
+  }
+
+  Widget _buildListTileDotsShape() {
+    return ListTile(
+      title: const Text('Dots shape: '),
+      trailing: DropdownButton<BoxShape>(
+        value: dotsShape,
+        items: BoxShape.values.map(_itemBuilderDotsShape).toList(),
+        onChanged: (value) {
+          setState(() {
+            dotsShape = value!;
+          });
+        },
+      ),
+    );
+  }
+
+  Widget _buildListTileDotsSpacing() {
+    return ListTile(
+      title: const Text('Dots spacing: '),
+      trailing: DropdownButton<EdgeInsets>(
+        value: dotsSpacing,
+        items: _kDotsSpacingMap.keys.map(_itemBuilderDotsSpacing).toList(),
+        onChanged: (value) {
+          setState(() {
+            dotsSpacing = value!;
+          });
+        },
+      ),
+    );
+  }
+
+  DropdownMenuItem<_DotsColor> _itemBuilderDotsColor(_DotsColor value) {
+    final text = value.toString().substring(value.toString().indexOf(".") + 1);
+    return DropdownMenuItem<_DotsColor>(
+      value: value,
+      child: Text(text),
+    );
+  }
+
+  DropdownMenuItem<double> _itemBuilderDotsDimen(double value) =>
+      DropdownMenuItem<double>(
+        value: value,
+        child: Text(value.toString()),
+      );
+
+  DropdownMenuItem<BoxShape> _itemBuilderDotsShape(BoxShape value) {
+    final text = value.toString().substring(value.toString().indexOf(".") + 1);
+    return DropdownMenuItem<BoxShape>(
+      value: value,
+      child: Text(text),
+    );
+  }
+
+  DropdownMenuItem<EdgeInsets> _itemBuilderDotsSpacing(String spacingKey) =>
+      DropdownMenuItem<EdgeInsets>(
+        value: _kDotsSpacingMap[spacingKey],
+        child: Text(spacingKey),
+      );
+
+  Color get getDotsColor => dotsColor.fold(
+        onPrimary: () => linden.colors.dotsIndicatorColor,
+        onSecondary: () => linden.colors.secondaryAccent,
+        onTertiary: () => linden.colors.primaryAction,
+      );
+
+  Color get getDotsActiveColor => dotsActiveColor.fold(
+        onPrimary: () => linden.colors.dotsIndicatorActiveColor,
+        onSecondary: () => linden.colors.secondaryAccent,
+        onTertiary: () => linden.colors.primaryAction,
+      );
+
+  Color get getDotsBorderColor => dotsBorderColor.fold(
+        onPrimary: () => linden.colors.dotsIndicatorBorderColor,
+        onSecondary: () => linden.colors.secondaryAccent,
+        onTertiary: () => linden.colors.primaryAction,
+      );
```

Comment:
💅 those looks exactly the same, so maybe we do not need an extension?
```dart
Color getColor(_DotsColor color){
  switch(color){
    case _DotsColor.primary:
        return linden.colors.dotsIndicatorBorderColor;
      case _DotsColor.secondary:
        return linden.colors.secondaryAccent;
      case _DotsColor.tertiary:
        return linden.colors.primaryAction;
  }
}
```

this one should be more then enough 😉

Reply:
Yeep, will do it, thanks!!

Reply:
Sorry, they are actually different for the primary color that might be color, activeColor and borderColor - but still I have an improvement that I'm going to try ;)